internal/security/fileprovider/tokenconfig_test.go

```diff
@@ -78,3 +78,138 @@ func TestLoadTokenConfigError2(t *testing.T) {
 	err := LoadTokenConfig(mockFileIoPerformer, "dummy-file", &tokenConf)
 	assert.Error(t, err)
 }
+
+func TestMergeWith(t *testing.T) {
+	map1 := TokenConfFile{
+		"key1": ServiceKey{},
+		"key2": ServiceKey{},
+	}
+
+	map2 := TokenConfFile{
+		"key3": ServiceKey{},
+		"key4": ServiceKey{},
+		"key5": ServiceKey{},
+	}
+
+	merged := map1.mergeWith(map2)
+
+	assert.Equal(t, 5, len(merged), "expect 5 entries in merged map")
+
+	// should get all keys in the merged map
+	expectedKeys := []string{"key1", "key2", "key3", "key4", "key5"}
+	for _, k := range expectedKeys {
+		assert.True(t, merged.keyExists(k))
+	}
+
+	// empty key case
+	empty := TokenConfFile{}
+	merged = map1.mergeWith(empty)
+	assert.Equal(t, map1, merged)
+
+	merged = empty.mergeWith(map1)
+	assert.Equal(t, map1, merged)
+
+	// some duplicate keys with other map
+	map3 := TokenConfFile{
+		"key1": ServiceKey{},
+		"key3": ServiceKey{},
+		"key7": ServiceKey{},
+	}
+
+	merged = map1.mergeWith(map3)
+
+	assert.Equal(t, 4, len(merged), "expect 4 entries in merged map")
+	expectedKeys = []string{"key1", "key2", "key3", "key7"}
+	for _, k := range expectedKeys {
+		assert.True(t, merged.keyExists(k))
+	}
+
+	merged = merged.mergeWith(map2)
```

Comment:
Is this the duplicates test? Make it its own test and initialize ServiceKey{} differently for each dup so you can validate you get the one you expect.

Reply:
yes, it is, i'll make it as own test

Reply:
We should also document the assumption being made here and why we are doing it the way we are doing it.

Reply:
added comments.